rpc/server.go

```diff
@@ -153,13 +153,13 @@ func (s *Server) Serve(handler http.Handler) {
 			defer s.mu.Unlock()
 
 			switch state {
-			case http.StateNew:
+			case http.StateNew, http.StateHijacked:
 				if s.closed {
 					conn.Close()
 					return
 				}
 				s.activeConns[conn] = struct{}{}
-			case http.StateHijacked, http.StateClosed:
+			case http.StateClosed:
```

Comment:
Yep! It wasn't until I read the handshake code in rpc/server.go that it became clear that RPC connections are indeed hijacked.